src/runtime/tensor.cpp

```diff
@@ -102,4 +103,69 @@ const Shape &Tensor::shape() const {
     return m_descriptor->m_shape;
 }
 
+namespace {
+template<class T, class Iterator>
+class View {
```

Comment:
It's definitely extra work, I believe we don't need `begin()`/`end()` for tensor